### DISCUSSION TITLE
Azure Datasource: Support Azure Billing API

### DISCUSSION THREAD

**Discussion by secustor** [grafana/grafana#41810] [Data Sources]
**What would you like to be added**:
I would like to display Azure cost metrics inside of Grafana. 

**Why is this needed**:
Grafana supports at the moment Azure Monitor, Application Insight and Log Analytics ( Kusto ).      
Adding the Billing API would be another step in the direction of full dashboarding capability for Azure.  

A specific use case is that we show metrics for technical personal already in Grafana, but Product Owners have to switch between Azure Portal for billing information and Grafana for technical insights.

**Comment by joshhunt** [comment]
I wonder if it's possible to query billing data from a Logs Workspace? 

I am a bit hesitant about adding _additional_ APIs to Azure Monitor, especially as we're engaging on a project to simplify the datasource, reduce the APIs it uses, and align it with Azure Monitor within Azure Portal. But I do appreciate how helpful it would be to have billing data available.

**Comment by secustor** [comment]
I have looked into it, but found no solution for that except writing some custom software which writes the metrics to logs and interpret it with Log Analytics or a  community Prometheus exporter

**Comment by ricohomewood** [comment]
There was a separate plug-in for this when azure monitor datasource was not part of the core grafana datasources. It was written as an installable plug-in by @yesoreyeram but I think has sine been deprecated? This was the linked issue: https://github.com/grafana/azure-monitor-datasource/issues/81

**Comment by darthkurak** [comment]
Azure costs in Grafana would be great. It would allows many useful insights and correlation with other metrics already existed. For example: costs per user/request

**Comment by al-lac** [comment]
We would also really like to have Azure costs in Grafana. Is the source code for the datasource of @yesoreyeram still available somewhere?

**Comment by billabongrob** [comment]
If powerBI can do this, certainly grafana can.  :) I'd love to see this, especially if sorting by tag.  @yesoreyeram code has gone missing and it was very helpful!

**Comment by Venture200** [comment]
hello, all. Is there any update or solution have you found out ?

**Comment by xhochn** [comment]
So Grafana can't handle the Azure Billing API? Does anyone have a solution/alternative?

**Comment by williambrach** [comment]
hello, is there any update or solution have you found?

**Comment by adyekjaer** [comment]
I did a bit of research and found that it was not that simple. To avoid writing too much code I saw two options:

For the ones running Grafana locally:
Simply install the data source using many of the forked repo's of the Azure Billing that went missing like: https://github.com/adyekjaer/grafana-azure-datasource

For the ones using Grafana cloud:
I went with the combination of running this project https://github.com/dodopizza/azure_billing_exporter
with a sidecar of prometheus in agent mode pushing metrics to my grafana cloud prometheus instance.

Drawback is that you don't easily have a nice way of showing history before it's been running for some time.
You can get azure_billing_exporter to create metrics based on e.g. resource group and or service names among others.

Hope this helps someone

**Comment by aangelisc** [comment]
Hi all,

Thanks for contributing to this discussion so far. I'd just like to understand if there's still interest in this feature request and get a better idea of what the ask is. My understanding is for Grafana to add support for displaying cost and usage information as shown in each Subscription in the Azure portal.

Please clarify this for me if I'm incorrect 😊

**Reply by adyekjaer** [reply]
Yes that's exactly right, basically a supported and maintained azure billing data source like: https://github.com/adyekjaer/grafana-azure-datasource

**Reply by darthkurak** [reply]
Also, confirm. We would like to get information which is available in https://learn.microsoft.com/en-us/azure/cost-management-billing/costs/quick-acm-cost-analysis

I think that this API: https://learn.microsoft.com/en-us/rest/api/cost-management/ is responsible for it.

**Reply by Strandfelt** [reply]
Also confirmed here. It would be a very nice addition to our Azure dashboards to be able to monitor the cost per subscription, resource group or resource.

**Reply by yggdrasil-tynor** [reply]
Chiming in, would be really awesome to have the ability to monitor cost and billing in the Azure Data Source plugin (which is already awesome)!

**Reply by Venture200** [reply]
A few months back, I managed to compose such dashboard by using Azure Cost Management API + Prometheus but the results were not accurate since Microsoft measures costs differently and metrics constantly changes till the end of billing period.

**Comment by Om-Amitesh** [comment]
Hello, Is there any update on this feature? I really need to integrate azure billing into grafana and display the montly usage on my dashboard. Im using azure managed grafana and was wondering if this is possible.

**Comment by aangelisc** [comment]
Thanks all for the continued response here! There's no update on this feature as of yet but as this discussion gains traction we will continue to evaluate and reprioritise future features!

**Reply by Venture200** [reply]
Looking forward for this feature !!!

**Reply by loopfish** [reply]
Thumbs up for this Costs Mgmt API to be accessible from Grafana please!

**Reply by saikrishna397** [reply]
@aangelisc Is there any update on this. This will really  be a good feature to have.

**Comment by rbharade** [comment]
I am not able to download file, is it still exists in repository ?

**Comment by peter-dolkens** [comment]
1 more voice looking for this feature - thanks

**Comment by ebuna** [comment]
+1 for this

**Comment by fclaudio-palmeira** [comment]
+1 here for this

**Comment by FloSchick** [comment]
+1 here for this

**Comment by iDugg** [comment]
+1 for this

**Comment by clin4** [comment]
+1 for this

**Comment by ghost** [comment]
+1 for this

**Comment by Ravio1i** [comment]
+1 for this

**Comment by raganar-ironside** [comment]
+1 for this

**Comment by gelicia** [comment]
Hello! We want to let you all know that we are moving to Github Issues for feature requests. You can read more about the move [here](https://github.com/grafana/grafana/discussions/73424).

This discussion was converted into an issue, which is available [here](https://github.com/grafana/grafana/issues/73443).
